evaluation error when inlining some symbols

Hi,

I got the following error:

```shell-session
$ OCAMLRUNPARAM=b dune exec -- owi c test/c/exit.c -vvv -w1
owi: [INFO] typechecking ...
owi: [INFO] linking      ...
owi: [INFO] interpreting ...
owi: [INFO] stack         : [  ]
owi: [INFO] running instr : call 11 (executed 0 times)
owi: [INFO] calling func  : func anonymous
owi: [INFO] stack         : [  ]
owi: [INFO] running instr : i32.const 0 (executed 0 times)
owi: [INFO] stack         : [ 0 ]
owi: [INFO] running instr : i32.const 0 (executed 0 times)
owi: [INFO] stack         : [ 0 ; 0 ]
owi: [INFO] running instr : call 10 (executed 0 times)
owi: [INFO] calling func  : func anonymous
owi: [INFO] stack         : [  ]
owi: [INFO] running instr : call 9 (executed 0 times)
owi: [INFO] calling func  : func anonymous
owi: [INFO] stack         : [  ]
owi: [INFO] running instr : block (executed 0 times)
owi: [INFO] stack         : [  ]
owi: [INFO] running instr : call 8 (executed 0 times)
owi: [INFO] calling func  : func anonymous
owi: [INFO] stack         : [  ]
owi: [INFO] running instr : call 2 (executed 0 times)
owi: [INFO] stack         : [ symbol_0 ]
owi: [INFO] running instr : local.tee 0 (executed 0 times)
owi: [INFO] stack         : [ symbol_0 ]
owi: [INFO] running instr : i32.eqz (executed 0 times)
owi: [INFO] stack         : [ (i32.of_bool (bool.eq symbol_0 0)) ]
owi: [INFO] running instr : br_if 0 (executed 0 times)
owi: [DEBUG] calling the solver
owi: [INFO] stack         : [  ]
owi: [INFO] running instr : local.get 0 (executed 0 times)
owi: [DEBUG] path condition smt query:
              (let-const symbol_0 i32)
              (assert (bool.eq symbol_0 (i32 0)))
              (check-sat)
owi: [INFO] stack         : [ symbol_0 ]
owi: [INFO] running instr : call 1 (executed 0 times)
owi: [DEBUG] path condition smt query:
              (let-const symbol_0 i32)
              (assert (bool.eq symbol_0 (i32 0)))
              (check-sat)
owi: [DEBUG] calling the solver
owi: [INFO] stack         : [  ]
owi: [INFO] running instr : i32.const 0 (executed 0 times)
owi: [DEBUG] path condition smt query:
              (let-const symbol_0 i32)
              (assert (bool.ne symbol_0 (i32 0)))
              (check-sat)
owi: [INFO] stack         : [ 0 ]
owi: [INFO] running instr : call 7 (executed 0 times)
owi: [DEBUG] path condition smt query:
              (let-const symbol_0 i32)
              (assert (bool.ne symbol_0 (i32 0)))
              (check-sat)
owi: [INFO] calling func  : func anonymous
owi: [INFO] stack         : [  ]
owi: [INFO] running instr : local.get 0 (executed 0 times)
owi: [DEBUG] path condition smt query:
              (let-const symbol_0 i32)
              (assert (bool.ne symbol_0 (i32 0)))
              (check-sat)
owi: [INFO] stack         : [ 0 ]
owi: [INFO] running instr : call 0 (executed 0 times)
owi: [DEBUG] path condition smt query:
              (let-const symbol_0 i32)
              (assert (bool.ne symbol_0 (i32 0)))
              (check-sat)
owi: [ERROR] a worker ended with exception Smtml.Eval.Eval_error(_), backtrace is: 
             Raised at Smtml__Eval.eval_error in file "src/smtml/eval.ml", line 67, characters 22-45
Called from Smtml__Expr.cvtop in file "src/smtml/expr.ml", line 509, characters 22-46
Called from Smtml__Expr.Set.map.(fun) in file "src/smtml/expr.ml", line 838, characters 18-23
Called from Stdlib__Set.Make.fold in file "set.ml", line 384, characters 34-55
Called from Owi__Symbolic_path_condition.slice_on_symbol in file "src/symbolic/symbolic_path_condition.ml", line 112, characters 6-57
Called from Owi__Symbolic_choice.check_reachability in file "src/symbolic/symbolic_choice.ml", line 154, characters 11-68
Called from Owi__State_monad.bind.(fun) in file "src/symbolic/state_monad.ml", line 34, characters 6-11
Called from Owi__State_monad.run in file "src/symbolic/state_monad.ml" (inlined), line 20, characters 26-32
Called from Owi__State_monad.bind.(fun) in file "src/symbolic/state_monad.ml", line 33, characters 19-28
Called from Owi__Scheduler.Schedulable.bind.(fun) in file "src/symbolic/scheduler.ml", line 15, characters 44-53
Called from Owi__Scheduler.Schedulable.bind.(fun) in file "src/symbolic/scheduler.ml", line 15, characters 44-53
Called from Owi__Scheduler.Schedulable.bind.(fun) in file "src/symbolic/scheduler.ml", line 15, characters 44-53
Called from Owi__Scheduler.Schedulable.bind.(fun) in file "src/symbolic/scheduler.ml", line 15, characters 44-53
Called from Owi__Scheduler.Schedulable.bind.(fun) in file "src/symbolic/scheduler.ml", line 15, characters 44-53
Called from Owi__Scheduler.Make.work.(fun) in file "src/symbolic/scheduler.ml", line 55, characters 33-39
Called from Stdlib__Fun.protect in file "fun.ml", line 34, characters 8-15
Re-raised at Stdlib__Fun.protect in file "fun.ml", line 39, characters 6-52
Called from Synchronizer.work_while in file "src/synchronizer.ml", lines 76-81, characters 4-31
Called from Owi__Scheduler.Make.run_worker.(fun) in file "src/symbolic/scheduler.ml", lines 63-65, characters 8-58

owi: [INFO] one domain exited with the following Smtml exception: The operator '`Cvtop (
            to_bool)' is not supported for type 'i32'
owi: [INFO] Completed paths: 0
All OK!
```

This was on https://github.com/OCamlPro/owi/pull/910

--- a/src/smtml/eval.ml
+++ b/src/smtml/eval.ml
@@ -737,8 +737,14 @@ module I32CvtOp = struct
     | Reinterpret_float -> bitv_of_int32 (of_fp32 1 op' v)
     | Sign_extend n -> to_bitv (Bitvector.sign_extend n (of_bitv 1 op' v))
     | Zero_extend n -> to_bitv (Bitvector.zero_extend n (of_bitv 1 op' v))
-    | OfBool -> v (* v is already a number here *)
-    | ToBool | _ -> eval_error (`Unsupported_operator (op', Ty_bitv 32))
+    | OfBool ->
+      let b = of_bool 1 (`Cvtop OfBool) v in
+      if b then to_bitv (Bitvector.make Z.one 32)
+      else to_bitv (Bitvector.make Z.zero 32)
+    | ToBool ->
+      let bv = of_bitv 1 (`Cvtop ToBool) v in
+      if not (Bitvector.eqz bv) then Value.True else False
+    | _ -> eval_error (`Unsupported_operator (op', Ty_bitv 32))
 end
 
 module I64CvtOp = struct

